opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/TracingFilter.java

```diff
@@ -134,19 +144,21 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         HttpServletResponse httpResponse = (HttpServletResponse) servletResponse;
 
         /**
-         * Check if request has been already traced do not proceed.
+         * If request is traced then do not start new span.
          */
         if (servletRequest.getAttribute(SERVER_SPAN_CONTEXT) != null) {
             chain.doFilter(servletRequest, servletResponse);
         } else if (isTraced(httpRequest, httpResponse)){
             SpanContext extractedContext = tracer.extract(Format.Builtin.HTTP_HEADERS,
                     new HttpServletRequestExtractAdapter(httpRequest));
 
-            Span span = tracer.buildSpan(httpRequest.getMethod())
+            final Span span = tracer.buildSpan(httpRequest.getMethod())
                     .asChildOf(extractedContext)
                     .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_SERVER)
                     .start();
 
+            final SpanWrapper spanWrapper = new SpanWrapper(span);
+            httpRequest.setAttribute(SERVER_SPAN, spanWrapper);
```

Comment:
nit: SERVER_SPAN_WRAPPER? I can imagine someone forgetting what they're supposed to cast to.